samples/matter/common/src/bridge/bridge_manager.cpp

```diff
@@ -307,30 +316,38 @@ CHIP_ERROR BridgeManager::AddDevices(MatterBridgedDevice *devices[], BridgedDevi
 				     uint8_t deviceListSize, chip::Optional<uint8_t> devicesPairIndexes[],
 				     uint16_t endpointIds[])
 {
-	VerifyOrReturnError(devices && dataProvider, CHIP_ERROR_INTERNAL);
+	CHIP_ERROR err{ CHIP_NO_ERROR };
+
+	VerifyOrReturnError(devices || dataProvider, CHIP_ERROR_INVALID_ARGUMENT);
+	VerifyOrExit(devices, err = CHIP_ERROR_INVALID_ARGUMENT);
+	VerifyOrExit(dataProvider, err = CHIP_ERROR_INVALID_ARGUMENT);
 
 	/* Maximum number of Matter bridged devices is controlled inside mDevicesMap,
 	   but the data providers may be created independently, so let's ensure we do not
 	   violate the maximum number of supported instances. */
-	if (mDevicesMap.FreeSlots() < deviceListSize) {
-		/* This method takes care of the resources, so objects have to be deleted. */
-		for (auto i = 0; i < deviceListSize; ++i) {
-			chip::Platform::Delete(devices[i]);
-		}
-		chip::Platform::Delete(dataProvider);
-		return CHIP_ERROR_NO_MEMORY;
-	}
+	VerifyOrExit(mDevicesMap.FreeSlots() >= deviceListSize, err = CHIP_ERROR_NO_MEMORY);
 
-	CHIP_ERROR status{ CHIP_NO_ERROR };
 	for (auto i = 0; i < deviceListSize; ++i) {
-		status = AddSingleDevice(devices[i], dataProvider, devicesPairIndexes[i], endpointIds[i]);
+		err = AddSingleDevice(devices[i], dataProvider, devicesPairIndexes[i], endpointIds[i]);
 
-		if (status != CHIP_NO_ERROR) {
-			return status;
+		if (err != CHIP_NO_ERROR) {
+			return err;
```

Comment:
In this case, shouldn't we release the devices not added so far? Or maybe remove those added and release all?

Reply:
Hmm yes, definitely the second option makes sense to me (though it's quite unluckily to hit it). @kkasperczyk-no any thoughts? I guess it's pointless to keep only the "part" of the bridged device when we fail to add the rest to the bridge (which would be the first Damian's suggestion).

Reply:
Yes, definitely. We should change it to `VerifyOrExit(err == CHIP_NO_ERROR)`

Reply:
It won't be that easy :) We need to take care about already added device and remove it explicitly from the `mDevicesMap`. Anyway, we are on the same page, so I'll try to implement the fix.

Reply:
Ok, I discovered some further limitations in the Bridge Manager, so I it's even more difficult that I initially thought. The good thing is that the case when we fail to add devices here is currently prevented by `VerifyOrExit(mDevicesMap.FreeSlots() >= deviceListSize, err = CHIP_ERROR_NO_MEMORY);` precondition. Let's close this PR and I will open up a follow up if I find the proper fix.